kotlin-native/litmus/src/commonMain/kotlin/WorkerTestRunner.kt

```diff
@@ -0,0 +1,62 @@
+import kotlinx.coroutines.runBlocking
+import kotlin.native.concurrent.TransferMode
+import kotlin.native.concurrent.Worker
+
+object WorkerTestRunner : LitmusTestRunner {
+    private data class WorkerContext(
+            val tests: List<BasicLitmusTest>,
+            val syncPeriod: Int,
+            val barrier: SpinBarrier,
+    )
+
+    override fun runTest(
+            batchSize: Int,
+            parameters: LitmusTestParameters,
+            testProducer: () -> BasicLitmusTest
```

Comment:
Why lambda here? Why not just a test itself (I guess you can implement `copy` method for tests or something like that)?